File: 1768-merge-strings-alternately/1768-merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        
        int n = word1.length();
        int m = word2.length();
        
        StringBuilder sb = new StringBuilder();
        
        int i = 0;
        while(i < n || i < m)
        {
            if(i < n)
                sb.append(word1.charAt(i));
            
            if(i < m)
                sb.append(word2.charAt(i));
            
            i++;
        }
        
        return sb.toString();
    }
}

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        n = len(word1)
        m = len(word2)
        
        sb = []
        i = 0;
        while i < n or i < m:
            if i < n:
                sb.append(word1[i])
                
            if i < m:
                sb.append(word2[i])    
                
            i += 1
            
        return ''.join(sb)